src/stan/io/json/json_data_handler.hpp

```diff
@@ -237,11 +237,8 @@ class json_data_handler : public stan::json::json_handler {
           vars_r[key] = pair;
         }
       } else {
-        if (!is_array_tuples(key_stack)) {
-          std::stringstream errorMsg;
-          errorMsg << "Attempt to redefine variable: " << key << ".";
-          throw json_error(errorMsg.str());
-        }
+        if (slot_types_map[key_stack[0]] != meta_type::ARRAY_OF_TUPLES)
+          unexpected_error(key);
```

Comment:
This change here seems to have broken tuples-of-array-of-tuples.

e.g., this Stan decl:

```stan
  tuple(array[2] real, int, array[3] tuple(real, array[4] int)) tuple_arr_tuple;
```

and this JSON

```json
{
  "tuple_arr_tuple": {
    "1": [ 5.6413957047783221, 6.86041453790074 ],
    "2": 3,
    "3": [
      { "1": 6.32749625894004, "2": [ 2, 3, 6, 4 ] },
      { "1": 3.3358848771578735, "2": [ 5, 3, 3, 2 ] },
      { "1": 2.3144782736186058, "2": [ 3, 4, 2, 5 ] }
    ]
  }
}
```


This previously worked but now raises `Variable tuple_arr_tuple.3.1 ill-formed data.`

Reply:
fixed, added to unit tests.